### DISCUSSION TITLE
subscribeProvider is missing in v5.1.3

### DISCUSSION THREAD

**Discussion by timl16** [reown-com/appkit#2763] [General]
This method subscribeProvider seems to be missing in the latest v5.1.13 and breaks the code. Can someone please take a look at this?

**Comment by timl16** [comment]
After calling createWeb3Modal, the returned object no longer includes the subscribeProvider method.

**Comment by enesozturk** [comment]
Hey @timl16, we recently made structural changes to AppKit and `subscribeProvider` is not part of the `createWeb3Modal` but we still expose the hook for subscribing to provider for React/Vue, like the following:

```
import { useWeb3ModalProvider, createWeb3Modal } from '@web3modal/wagmi/react'
// or
import { useWeb3ModalProvider, createWeb3Modal } from '@web3modal/wagmi/vue'
```

For plain JS we could simply export directly the `subscribeProvider` as a function from the package. Could you share which environment (JS/React/Vue, etc) and client (wagmi/ethers/ethers5) you're using please?

**Reply by timl16** [reply]
@enesozturk Thanks a lot for your reply. I'm using plain JS and look forward to your direct export of the subscribeProvider function.

**Reply by timl16** [reply]
I am using ethers.

**Reply by ms-safiq** [reply]
I am using js ethers. How do i use subscribeProvider? @enesozturk

**Reply by timl16** [reply]
> I am using js ethers. How do i use subscribeProvider? @enesozturk

I am currently using version 5.0.6, where the subscribeProvider method is still available. I will upgrade to the latest version once this method is exported, as @enesozturk  mentioned in the previous post.

**Reply by enesozturk** [reply]
Hey @timl16 @ms-safiq , the missing methods are brought back in the latest versions. Please give the `5.1.6` a try: 
- https://www.npmjs.com/package/@web3modal/ethers/v/5.1.6

Also to see docs of the methods and listeners, visit: https://docs.walletconnect.com/appkit/javascript/core/actions

**Reply by timl16** [reply]
@enesozturk Cool! I tested version 5.1.6 today, and it worked perfectly. Thank you so much for your quick turnaround.

**Reply by ms-safiq** [reply]
@rtomas @timl16  How do we disconnect? There is notdiconnect method like before.

**Reply by ms-safiq** [reply]
I am using javascript version ethers

**Reply by rtomas** [reply]
Hi @ms-safiq, we’ll address this ASAP. In the meantime, you can use an older version or use `a.adapters![0].connectionControllerClient?.disconnect`
